File: VB/WindowsFormsApplication25/RepositoryItemCustomBreadCrumbEdit.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository

Namespace WindowsFormsApplication25
    <UserRepositoryItem("RegisterCustomBreadCrumbEdit")> _
    Public Class RepositoryItemCustomBreadCrumbEdit
        Inherits RepositoryItemBreadCrumbEdit

        Shared Sub New()
            RegisterCustomBreadCrumbEdit()
        End Sub

        Public Const CustomEditName As String = "CustomBreadCrumbEdit"

        Public Sub New()
        End Sub

        Public Overrides ReadOnly Property EditorTypeName() As String
            Get
                Return CustomEditName
            End Get
        End Property

        Public Shared Sub RegisterCustomBreadCrumbEdit()
            Dim img As Image = Nothing
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(CustomEditName, GetType(CustomBreadCrumbEdit), GetType(RepositoryItemCustomBreadCrumbEdit), GetType(CustomBreadCrumbEditViewInfo), New BreadCrumbEditPainter(), True, img))
        End Sub
        Private _allowDropDownButtonSelectedState As Boolean

        Public Property AllowDropDownButtonSelectedState() As Boolean
            Get
                Return _allowDropDownButtonSelectedState
            End Get
            Set(ByVal value As Boolean)
                If _allowDropDownButtonSelectedState <> value Then
                    _allowDropDownButtonSelectedState = value
                    OnPropertiesChanged()
                End If
            End Set
        End Property
        Public Overrides Sub Assign(ByVal item As RepositoryItem)
            BeginUpdate()
            Try
                MyBase.Assign(item)
                Dim source As RepositoryItemCustomBreadCrumbEdit = TryCast(item, RepositoryItemCustomBreadCrumbEdit)
                If source Is Nothing Then
                    Return
                End If
                _allowDropDownButtonSelectedState = source.AllowDropDownButtonSelectedState
            Finally
                EndUpdate()
            End Try
        End Sub
    End Class

End Namespace

File: CS/WindowsFormsApplication25/RepositoryItemCustomBreadCrumbEdit.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;

namespace WindowsFormsApplication25 {
    [UserRepositoryItem("RegisterCustomBreadCrumbEdit")]
    public class RepositoryItemCustomBreadCrumbEdit : RepositoryItemBreadCrumbEdit {
        static RepositoryItemCustomBreadCrumbEdit() {
            RegisterCustomBreadCrumbEdit();
        }

        public const string CustomEditName = "CustomBreadCrumbEdit";

        public RepositoryItemCustomBreadCrumbEdit() {
        }

        public override string EditorTypeName {
            get {
                return CustomEditName;
            }
        }

        public static void RegisterCustomBreadCrumbEdit() {
            Image img = null;
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(CustomEditName, typeof(CustomBreadCrumbEdit), typeof(RepositoryItemCustomBreadCrumbEdit), typeof(CustomBreadCrumbEditViewInfo), new BreadCrumbEditPainter(), true, img));
        }
        private bool _allowDropDownButtonSelectedState;

        public bool AllowDropDownButtonSelectedState {
            get {
                return _allowDropDownButtonSelectedState;
            }
            set {
                if(_allowDropDownButtonSelectedState != value) {
                    _allowDropDownButtonSelectedState = value;
                    OnPropertiesChanged();
                }
            }
        }
        public override void Assign(RepositoryItem item) {
            BeginUpdate();
            try {
                base.Assign(item);
                RepositoryItemCustomBreadCrumbEdit source = item as RepositoryItemCustomBreadCrumbEdit;
                if(source == null)
                    return;
                _allowDropDownButtonSelectedState = source.AllowDropDownButtonSelectedState;
            } finally {
                EndUpdate();
            }
        }
    }

}
